gcexport.py

```diff
@@ -216,34 +245,58 @@ def http_req(url, post=None, headers=None):
         logging.info('Got 204 for %s, returning empty response', url)
-        return ''
+        return b''
     elif response.getcode() != 200:
         raise Exception('Bad return code (' + str(response.getcode()) + ') for: ' + url)
 
     return response.read()
 
+def http_req_as_string(url, post=None, headers=None):
+    """Helper function that makes the HTTP requests, returning a string instead of bytes."""
+    if python3:
+        return http_req(url, post, headers).decode()
+    else:
+        return http_req(url, post, headers)
+
 
 # idea stolen from https://stackoverflow.com/a/31852401/3686
-def load_properties(multiline, sep='=', comment_char='#', keys=None):
+def load_properties(multiline, separator='=', comment_char='#', keys=None):
     """
-    Read a multiline string of properties (key/value pair separated by *sep*) into a dict
+    Read a multiline string of properties (key/value pair separated by *separator*) into a dict
 
     :param multiline:    input string of properties
-    :param sep:          separator between key and value
+    :param separator:    separator between key and value
     :param comment_char: lines starting with this char are considered comments, not key/value pairs
     :param keys:         list to append the keys to
     :return:
     """
     props = {}
     for line in multiline.splitlines():
         stripped_line = line.strip()
-        if stripped_line and not stripped_line.startswith(comment_char):
-            key_value = stripped_line.split(sep)
-            key = key_value[0].strip()
-            value = sep.join(key_value[1:]).strip().strip('"')
-            props[key] = value
-            if keys != None:
-                keys.append(key)
+        if python3:
+            stripped_line = line.strip()
+            if stripped_line:
+                if (isinstance(stripped_line,bytes)):
+                    stripped_line = stripped_line.decode('utf8')
+                if (isinstance(comment_char,bytes)):
+                    comment_char=comment_char.decode('utf8')             
+                if not stripped_line.startswith(comment_char):
+                    #https://python-forum.io/Thread-Diff-between-Py-2-7-and-3--14749
+                    key_value = stripped_line.split(separator)
+                    key = key_value[0].strip()
+                    value = separator.join(key_value[1:]).strip().strip('"')
+                    props[key] = value
+                    if keys != None:
+                        keys.append(key)
+        else:
+            stripped_line = line.strip()
+            if stripped_line and not stripped_line.startswith(comment_char):
+                key_value = stripped_line.split(separator)
+                key = key_value[0].strip()
+                value = separator.join(key_value[1:]).strip().strip('"')
+                props[key] = value
+                if keys != None:
+                    keys.append(key)
```

Comment:
What exactly is happening in this part? This block of code looks more complex than probably it should to be. I may comment some single lines but honestly, I'm pretty sure it should be refactored first or some general comment should be added on the top of it.

Reply:
IIUC the special code was introduced because at the beginning of the migration to Py 3 the properties files were read as bytes instead of strings. This has been fixed some days ago and I just removed the special code, it should not be needed anymore.